src/libra_py/workflows/qsh/qstochastic_Ham.py

```diff
@@ -244,50 +224,37 @@ def run(params):
         H_vib_re.append(hvib_re)
         H_vib_im.append(hvib_im)
 
-        dE = MATRIX(nstates, nstates)
-        for a in xrange(nstates):
-            for b in xrange(nstates):
-                dE.set(a,b, hvib_re.get(a,a) - hvib_re.get(b,b))
-        dH.append(dE)
-
-
     
-    H_vib_re_ave, H_vib_re_std, dw_Hvib_re, up_Hvib_re = mat_stat(H_vib_re)
-    H_vib_im_ave, H_vib_im_std, dw_Hvib_im, up_Hvib_im = mat_stat(H_vib_im)    
+    H_vib_re_ave, H_vib_re_std, dw_Hvib_re, up_Hvib_re = comn.mat_stat(H_vib_re)
+    H_vib_im_ave, H_vib_im_std, dw_Hvib_im, up_Hvib_im = comn.mat_stat(H_vib_im)    
     print "dw_re = "; dw_Hvib_re.show_matrix()
     print "up_re = "; up_Hvib_re.show_matrix()
     print "dw_im = "; dw_Hvib_im.show_matrix()
     print "up_im = "; up_Hvib_im.show_matrix()
 
 
-    freqs_re0 = mat_freqs(H_vib_re, 0, 0, dt, "H_vib_re_E0_", Nfreqs)
-    freqs_re1 = mat_freqs(H_vib_re, 1, 1, dt, "H_vib_re_E1_", Nfreqs)
-    freqs_im = mat_freqs(H_vib_im, 0, 1, dt, "H_vib_im_D01_", Nfreqs)
-
-    dH_ave, dH_std, dw_dH, up_dH =  mat_stat(dH)
+    freqs_re0 = mat_freqs(H_vib_re, 0, 0, dt, "_H_vib_re_E0_", Nfreqs)

```

Comment:
we need to generalize this and other sections to the case when we have more than 2 state involved